src/main/java/Entities/Group.java

```diff
@@ -0,0 +1,89 @@
+package main.java.Entities;
+import main.java.Entities.User;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+
+public class Group {
+    /** Creates a Group object */
+    private final String groupName;
+    private String groupCode;
+    private final ArrayList<String> groupMembers;
+    private final ArrayList<String> usedCodes;
+    private final RecipeBook recipeBook;
+
+
+    /**
+     * Creates a group with a name, group code, group members, and Recipe Book.
+     * @param groupName - the name of the group
+     */
+    public Group(String groupName) {
+        this.groupName = groupName;
+        this.groupCode = "";
```

Comment:
I'm not sure, but this looks dangerous: it is possible to initialize elsewhere in the code a group without a group code.